Kernel/Devices/Storage/USB/USBStorageController.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2023, Leon Albrecht <leon.a@serenityos.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/IntrusiveList.h>
+#include <Kernel/Devices/Storage/StorageController.h>
+#include <Kernel/Devices/Storage/StorageDevice.h>
+#include <Kernel/Devices/Storage/StorageManagement.h>
+#include <Kernel/Devices/Storage/USB/MassStorageDevice.h>
+
+namespace Kernel::USB {
+
+class StorageController : public ::StorageController {
+public:
+    static StorageController& the();
+
+    virtual LockRefPtr<StorageDevice> device(u32 index) const override
+    {
+        // FIXME: Avoid this const_cast
+        return const_cast<Vector<NonnullLockRefPtr<BulkSCSIInteface>>&>(m_devices).at(index);
+    }
+    virtual size_t devices_count() const override
+    {
+        dbgln("Query USB Storage Device Count");
+        return m_devices.size();
+    }
+
+    void register_device(BulkSCSIInteface& device)
+    {
+        m_devices.append(device);
+        StorageManagement::the().add_device(device);
+    }
+    void remove_device(BulkSCSIInteface& device)
+    {
+        StorageManagement::the().remove_device(device);
+        auto index = m_devices.find_first_index(device);
+        VERIFY(index.has_value());
+        m_devices.remove(index.value());
```

Comment:
Maybe `m_devices.remove_first_matching(..)`?

Reply:
That one only comes with a predicate, so I thought that this was simpler than going through a lambda

Reply:
Sure, it's just a preference of style. At least you can drop the `VERIFY` since that is already done by `index.value()`.

Reply:
I think we had a discussion about that once on discord, that we always want to check optionals before releasing them, so I'd like to leave it there
It also makes the Error message a tiny bit nicer